### ISSUE TITLE
PST to PDT is off by one day

### ISSUE THREAD

**Issue by ricardobanegas** [spencermountain/spacetime#44]
Hello all. 

I have a method that converts UTC to a dynamic local time but testing using `America/Los_Angeles` the library seems to be off by one day when **PST** goes over to **PDT** (winter to summer time)

According to [timeanddate.com/time/zone/usa/los-angeles](https://www.timeanddate.com/time/zone/usa/los-angeles) the time change should occur `2018-03-11 02:00:00` but this seems to happen `2018-03-12 02:00:00` local time. 

**Code**
```javascript

// From config config.TIME.FORMAT.RAW_DATE_UNIX_LOCAL
RAW_DATE_UNIX_LOCAL: 'YYYY-MM-dd HH:mm:ss'

/**
 * Converts from UTC to naive time - timezone aware
 * @param {string} date
 */
function _dateUTCToLocal (date) {
    // Display as local time
    const s =
        spacetime(date)
        .goto(store.getters.getUser.facility.timezone)

    console.log('BEFORE', date, 'AFTER', s.format(config.TIME.FORMAT.RAW_DATE_UNIX_LOCAL))

    // Display as local time
    return s.format(config.TIME.FORMAT.RAW_DATE_UNIX_LOCAL)
}

```

**`console.log` output**
```
BEFORE 2018-03-12T07:00:00+00:00 AFTER 2018-03-11 23:00:00
BEFORE 2018-03-12T08:00:00+00:00 AFTER 2018-03-12 00:00:00
BEFORE 2018-03-12T09:00:00+00:00 AFTER 2018-03-12 01:00:00
BEFORE 2018-03-12T10:00:00+00:00 AFTER 2018-03-12 03:00:00
BEFORE 2018-03-12T11:00:00+00:00 AFTER 2018-03-12 04:00:00
BEFORE 2018-03-12T12:00:00+00:00 AFTER 2018-03-12 05:00:00
```

Thanks in advance. Love the project!

**Comment by spencermountain** [comment]
hey Ricardo! yeah, nice find. I think we're still using 2017's dst dates.
rats, i gotta do that.
i'll try to get this today, or this at least week.
thanks for finding this.
cheers

**Comment by ricardobanegas** [comment]
@spencermountain 

My pleasure. Let me know if you need help testing.

**Comment by spencermountain** [comment]
hey @ricardobanegas , wanna check out `v3.2.0`?
this is our first year-update, so may find some funny-biz.
https://runkit.com/spencermountain/5a84aad980921200129df457
cheers

**Comment by ricardobanegas** [comment]
@spencermountain Initial tests look good! Will place code on staging and let our tester run through some deeper tests. I'll let you know.

**Comment by ricardobanegas** [comment]
@spencermountain Sorry, I spoke to soon. There is still a bug in the conversions. 

On `2018-03-11` I expect `12:00:00 to 14:00:00` to be the converted values, but I'm getting `11:00:00 to 13:00:00`. 

It seems the library understand the offset is -7 in the ` s.timezone().current` object but not when doing the calculation.

This is produced by same code snippet above.

```javascript
BEFORE 2018-03-11T19:00:00+00:00 AFTER 2018-03-11 11:00:00 {offset: -7, isDST: true}
BEFORE 2018-03-11T20:00:00+00:00 AFTER 2018-03-11 12:00:00 {offset: -7, isDST: true}
BEFORE 2018-03-11T21:00:00+00:00 AFTER 2018-03-11 13:00:00 {offset: -7, isDST: true}
```

Thanks again.

**Comment by ricardobanegas** [comment]
Another bug on the same day. In the beginning of the day its returning -9 in offset. It seems the entire day `2018-03-11` is off by one hour.

```javascript
BEFORE 2018-03-11T09:00:00+00:00 AFTER 2018-03-11 00:00:00 {offset: -8, isDST: false}
BEFORE 2018-03-11T10:00:00+00:00 AFTER 2018-03-11 01:00:00 {offset: -8, isDST: false}
BEFORE 2018-03-11T11:00:00+00:00 AFTER 2018-03-11 03:00:00 {offset: -7, isDST: true}
BEFORE 2018-03-11T12:00:00+00:00 AFTER 2018-03-11 04:00:00 {offset: -7, isDST: true}
```

**Comment by dekoza** [comment]
A tip that might be helpful: [pendulum](https://github.com/sdispater/pendulum) for Python makes perfect calculations regarding timezones and DST. Maybe you could extract that algorithm and put it in `spacetime`. Although it makes use of very precise (zoneinfo database)[https://github.com/sdispater/pytzdata/tree/master/pytzdata/zoneinfo].

**Comment by spencermountain** [comment]
hey @ricardobanegas - just confirming, this is for `America/Los_Angeles` on the day of the dst change right?
Yeah, i think the dst-change we have is an hour late. Would that make sense? I think it's a "3am becomes 2am" confusion on my part.

@dekoza ohh pendulum is really cool! thank you for sharing that.

yeah, we're definitely at the point in this project where we can start diffing results with other libraries to find the sore-spots in ours. Been shy about doing that so far.

**Comment by ricardobanegas** [comment]
@spencermountain corrects its America/Los_Angeles we are talking about.

**Comment by spencermountain** [comment]
hey, good find. looks like all the dst-changes were off by one hour.
fixed in `v4.0.0` - wanna check it out?
thanks
